core/tests/puppeteer/lighthouse_setup.js

```diff
@@ -274,26 +274,10 @@ const generateDataForTopicAndStoryPlayer = async function(browser, page) {
   try {
     await page.goto('http://127.0.0.1:8181/admin#/activities', { waitUntil: networkIdle });
 
-    await page.waitForSelector('oppia-admin-dev-mode-activities-tab');
-    await page.evaluate(async () => {
-      const adminPageComponent = document.querySelector('oppia-admin-dev-mode-activities-tab');
+    await page.waitForSelector(generateTopicButton);
+    await page.click(generateTopicButton);
 
-      // Check if the component is found
-      if (!adminPageComponent) {
-        throw new Error('Admin page not found');
-      }
-
-      await adminPageComponent.loadNewStructuresData();
```

Comment:
Don't load data structures using `adminPageComponent.loadNewStructuresData`. In an E2E test, you interact with the web page and see the results as if you were the user. 

Try this: click on the generateTopicButton as you already are currently doing. We set a status message for the end user ([here](https://github.com/oppia/oppia/blob/dd4e4defab5abbba17bbc1d99f406d850ea0f68c/core/templates/pages/admin-page/activities-tab/admin-dev-mode-activities-tab.component.ts#L151)) that is rendered on the web page. See if you can find the element that is rendering that message and wait until that element and the exact message have both appeared.